omega/omega-transaction/src/main/java/org/apache/servicecomb/pack/omega/transaction/TransactionAspect.java

```diff
@@ -59,4 +70,15 @@ Object advise(ProceedingJoinPoint joinPoint, Compensable compensable) throws Thr
       LOG.debug("Restored context back to {}", context);
     }
   }
+
+  TransactionContext getTransactionContextFromArgs(Object[] args) {
+    if (args != null) {
+      for (Object arg : args) {
+        if (arg instanceof TransactionContextWrapper) {
+          return ((TransactionContextWrapper) arg).getTransactionContext();
+        }
```

Comment:
I think we can add one more detection:

```java
else if (arg instanceof TransactionContext) {
  return ((TransactionContextWrapper) arg);
}
```

Reply:
Do you mean that user could just pass the TransactionContext as a parament to the subtransaction class?  Yeah, we could support this feature at the meantime.